docs/architecture/flux.md

```diff
@@ -0,0 +1,77 @@
+# Firefox Lockbox for Android Architecture
+
+## RxKotlin
+
+Firefox Lockbox for Android makes extensive use of RxKotlin, an implementation of the Observable pattern from ReactiveX. More information and many marble diagrams can be found in the [ReactiveX documentation](http://reactivex.io/). The rest of this document relies on a basic understanding of the reader of the ReactiveX-style Observer implementation. Their intro document is a [good starting point](http://reactivex.io/intro.html).
+
+## Flux
+
+### Architecture Pattern
+
+In short, Flux architecture design maintains a unidirectional data flow, in which a global Dispatcher receives Actions & dispatches them to appropriate Stores. The Stores, in turn, process data & provide the source of truth for the Views. As users interact with the Views, any updates are made via a dispatched Action and the cycle begins again. See this [flux architecture](https://facebook.github.io/flux/docs/overview.html) writeup for more details on the original Flux architecture scheme.
+
+Lockbox implements a modified version of the described architecture (LockFlux), keeping in mind that the original implementation ignores asynchronous work. In this implementation, all asynchronous work is handled by the Stores as they reduce actions and state updates to their observable state.
+
+### Memory Management
+
+The five major components of this architecture (`View`, `Presenter`, `Store`, `Dispatcher`, and `Action`) have distinct lifecycle management based on their functions.
+
+`View`/`Presenter` pairs are allocated and de-allocated as views get displayed or hidden in turn.
+
+`Store`s and the `Dispatcher` are global singleton objects, meaning that they get lazy-loaded by the application as their shared members get accessed by the `Presenter`s for view configuration or dispatching.
+
+`Action`s get deallocated as soon as they reach the end observer for their intended function.
+
+### View/Presenter
```

Comment:
Somewhere there needs to be the call out that (unless there's a really good reason otherwise), the `View` is a `Fragment` and not an `Activity`.  It might just be me, but it seems that Activities are traditionally seen as the View, and Lockbox is going in a different direction.

Reply:
done.